src/pytom_tm/extract.py

```diff
@@ -11,29 +11,113 @@
 from pytom_tm.angles import load_angle_list, convert_euler
 from pytom_tm.io import read_mrc
 from scipy.special import erfcinv
+from scipy.optimize import curve_fit
 from tqdm import tqdm
 
 
-def detect_blobs(
-        volume: npt.NDArray[float],
-        voxel_size: float,
-        blob_size: float,
-        threshold: float = 1.
-) -> npt.NDArray[int]:
-    blob_radius = (blob_size / voxel_size) / 2
-    sigma = blob_radius / np.sqrt(3)
-    filtered = ndimage.gaussian_laplace((volume - volume.mean()) / volume.std(), sigma)
-    return ((filtered > threshold) * 1).astype(int)
+plotting_available = False
+try:
+    import matplotlib
+    import matplotlib.pyplot as plt
+    import seaborn as sns
+    sns.set(context='talk', style='ticks')
+    plotting_available = True
+except ModuleNotFoundError:
+    pass
+
+
+def predict_tophat_mask(
+        score_volume: npt.NDArray[float],
+        output_path: Optional[pathlib.Path] = None,
+        n_false_positives: int = 1,
+) -> npt.NDArray[bool]:
+    tophat = ndimage.white_tophat(
+        score_volume,
+        structure=ndimage.generate_binary_structure(
+            rank=3,
+            connectivity=1
+        )
+    )
+    y, bins = np.histogram(tophat.flatten(), bins=50)
+    x_raw, y_raw = (bins[2:-1] + bins[3:]) / 2, y[2:]  # discard first two bin because of over-representation of zeros
```

Comment:
```suggestion
    bin_centers = (bins[:-1] + bins[1:]) / 2
    x_raw, y_raw = bin_centers[2:], y[2:]  # discard first two points because of over-representation of zeros
```